e2e/helpers/tables.ts

```diff
@@ -0,0 +1,420 @@
+    const config = columnsSchemaConfig[index]
+    const cell = tableCells.nth(index + 1) // shift right for checkboxes
+    const value = tableData[config.name]
+
+    if (config.type === 'Date') {
+      await selectTableDate(page, cell, value as string)
+    } else if (config.type.includes('List')) {
+      await enterTableListValues(page, cell, value)
+    } else if (config.restrictValues || config.type === 'Boolean') {
+      await selectTableRestrictValues(cell, value as string)
+    } else {
+      await enterTableValue(cell, value)
+    }
+  }
+}
+
+export const enterTableData = async (
+  page: Page,
+  columnsSchemaConfig: ColumnSchemaConfig[],
+  tableData: TableDataConfig[],
+) => {
+  const tableRows = await getTableEditorRows(page)
+
+  // skip initial empty column
+  await expectTableColumnHeaders(tableRows, columnsSchemaConfig, 1)
+
+  await test.step('enter test data', async () => {
+    const addRowButton = page.getByRole('button', { name: 'Add Row' })
+    for (let index = 0; index < tableData.length; index++) {
+      await test.step(`enter test data: ${tableData[index].item}`, async () => {
+        await addRowButton.click()
+
+        // start from 1 so column headers in first row are skipped
+        const cells = tableRows.nth(index + 1).getByRole('cell')
+        await enterTableDataRow(
+          page,
+          cells,
+          columnsSchemaConfig,
+          tableData[index],
+        )
+      })
+    }
+  })
+}
+
+// Enter individual table cells ------------------------------------------------
+// non-list type, no restricted value
+export const enterTableValue = async (
+  cell: Locator,
+  value: string | number | string[],
+) => {
+  await cell.locator('.form-control').fill(value.toString())
+}
+
+const getDecadeIndex = (year: number) => Math.floor(year / 10)
+const dateStringToDate = (dateString: string) => {
+  return new Date(`${dateString}T12:00:00.000Z`)
+}
+
+export const selectTableDate = async (
+  page: Page,
+  cell: Locator,
+  dateString: string,
+) => {
+  await test.step('select table date', async () => {
+    const date = dateStringToDate(dateString)
+    const year = date.getFullYear()
+    const decade = getDecadeIndex(year)
+    const currentDecade = getDecadeIndex(new Date().getFullYear())
+    const decadeDifference = currentDecade - decade
+
+    const month = date.toLocaleString('default', { month: 'short' })
+    const day = date.getDate()
+
+    await cell.click()
+    const switchButton = page.locator('.switch:visible')
+
+    await test.step('select year', async () => {
+      await switchButton.click()
+      await switchButton.click()
+
+      if (decadeDifference !== 0) {
+        await test.step('change decade', async () => {
+          const arrowSelector =
+            decadeDifference > 0 ? '.prev:visible' : '.next:visible'
+          for (let i = 0; i < decadeDifference; i++) {
+            await page.locator(arrowSelector).click()
+          }
+        })
+      }
```

Comment:
Sorry, what's going on here? Not clear to me why we're changing the decade (conditionally?)

Reply:
That makes sense! It makes it super-resilient. My suggestion is to consider encapsulating this logic in a function that is dedicated to picking a date specifically using the `goog.ui.DatePicker` implementation, then just call that here.

Reply:
Good idea! Will do